suprb/optimizer/rule/es/es.py

```diff
@@ -100,36 +121,22 @@ def _optimize(self, X: np.ndarray, y: np.ndarray, initial_rule: Rule, random_sta
                 warnings.warn("No valid children were generated during this iteration.", UserWarning)
                 continue
 
-            # For CSR only the spread mutation rate gets altered, for MPR both rate are altered
+            """Use the adapted version of the 1/5th-Rule as introduced in:
+               https://direct.mit.edu/evco/article-abstract/9/2/147/898/Convergence-in-Evolutionary-Programs-with-Self 
+               proportion: float 
+                    How many of the lmbda children create are fitter than their parent
+            """
             if self.adaptive_sigma:
                 matching_type = elitist.match.__class__.__name__
                 fitter_children = sum([1 for child in children if child.fitness_ > elitist.fitness_])
                 proportion = fitter_children / self.lmbda
 
                 if proportion > 0.2:
-                    if matching_type in ("OrderedBound", "UnorderedBound"):
-                        self.mutation.sigma /= 0.85
-                    elif matching_type == "CentreSpread":
-                        self.mutation.sigma[1] /= 0.85
-                    elif matching_type == "MinPercentage":
-                        self.mutation.sigma[0] /= random_state.normal(loc=0.85, scale=0.01)
-                        self.mutation.sigma[1] /= 0.85
+                    self.alter_sigma(self.adaption_rate, "/", matching_type, random_state)
                 elif 0.05 <= proportion < 0.2:
-                    if matching_type in ("OrderedBound", "UnorderedBound"):
-                        self.mutation.sigma *= 0.85
-                    elif matching_type == "CentreSpread":
-                        self.mutation.sigma[1] *= 0.85
-                    elif matching_type == "MinPercentage":
-                        self.mutation.sigma[0] *= random_state.normal(loc=0.85, scale=0.01)
-                        self.mutation.sigma[1] *= 0.85
+                    self.alter_sigma(self.adaption_rate, "*", matching_type, random_state)
                 elif proportion < 0.05:
-                    if matching_type in ("OrderedBound", "UnorderedBound"):
-                        self.mutation.sigma *= 2
-                    elif matching_type == "CentreSpread":
-                        self.mutation.sigma[1] *= 2
-                    elif matching_type == "MinPercentage":
-                        self.mutation.sigma[0] *= random_state.normal(loc=2, scale=0.01)
-                        self.mutation.sigma[1] *= 2
+                    self.alter_sigma(2, "*", matching_type, random_state)
```

Comment:
Can you please add a short comment why we are doing this here? (To move faster into "good" regions if we produce to few better rules)